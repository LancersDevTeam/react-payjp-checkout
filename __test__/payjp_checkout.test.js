import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson,  mountToJson} from 'enzyme-to-json';
import PayjpCheckout from '../src/payjp_checkout';

describe('PayjpCheckout', ()=>{
    test('renders correctly', ()=>{
        const payjpCheckoutProps = {
            // dataKey は各自の公開鍵に置き換えて下さい
            dataKey: 'pk_test_0383a1b8f91e8a6e3ea0e2a9',
            dataText: 'クレジットカードで支払う',
            dataPartial: 'true',
            onCreatedHandler: () => { console.log('onCreatedHandler'); },
            onFailedHandler: () => { console.log('onFailedHandler'); },
        }
        const tree = mountToJson(mount(
            <PayjpCheckout {...payjpCheckoutProps} />
        ));
        expect(tree).toMatchSnapshot();
    });
});
