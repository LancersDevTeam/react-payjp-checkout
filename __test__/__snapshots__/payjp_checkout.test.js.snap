// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PayjpCheckout renders correctly 1`] = `
<PayjpCheckout
  className="payjp-button"
  dataKey="pk_test_0383a1b8f91e8a6e3ea0e2a9"
  dataPartial="true"
  dataText="クレジットカードで支払う"
  onCreatedHandler={[Function]}
  onFailedHandler={[Function]}
>
  <div />
</PayjpCheckout>
`;
